util/src/main/java/com/spotify/hamcrest/util/DescriptionUtils.java

```diff
@@ -77,18 +77,18 @@ public static void indentDescription(Description description, Description innerD
    *                       which will write the describe the mismatch for that key
    * @param describeKey A {@link BiConsumer} used to describe the key
    */
-  public static void describeNestedMismatches(
-      Set<String> allKeys,
+  public static <T> void describeNestedMismatches(
```

Comment:
That's exactly what I had to go from in the latest commit:
https://github.com/spotify/java-hamcrest/pull/19/commits/58bf1f3206a671f5f1f6c767729bf51d73418f98

Strings will make multiple identical matches like following impossible, which means we have to throw exception when someone actually tries that:
https://github.com/spotify/java-hamcrest/pull/19/files#diff-a284c930d1a406d7ffee257ebf1b9477R231